src/Health/AbstractHealth.php

```diff
@@ -0,0 +1,45 @@
+<?php
+declare(strict_types=1);
+
+namespace EoneoPay\Externals\Health;
+
+use EoneoPay\Externals\Health\Interfaces\HealthInterface;
+
+abstract class AbstractHealth implements HealthInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    abstract public function getChecks(): array;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function extended(): array
+    {
+        // Get the classes to call upon for health checks
+        $checks = $this->getChecks();
+
+        // If there are no checks set, return early.
+        if (\count($checks) === 0) {
+            return [];
```

Comment:
`self::HEALTHY` ?

Reply:
The return type is `array` for this method that returns either `HEALTHY` or `DEGRADED` for each check. If there are no checks performed, the resulting array is empty because there is nothing to check?

Reply:
I think the actual extended output should contain an overall health property with an array of individual results, which'd fix this issue allowing you to return a overall status value

Reply:
sounds like a plan